### PR TITLE
Track CompoundType and Buffer only

### LINKED ISSUE #481: v3.3.0 need more than 30 hours for syncing mainnet even it's just about 1.7million blocks
Syncing mode: offline import data from chain.0.acc
CPU: I7-10875H, 8 cores, 16 processors
Disk: 2TB SSD
RAM: 48GB
Cost time: More than 30 hours
Due to our investigation and several tests, something extremely lower the speed 5x ~ 10x for execution from https://github.com/neo-project/neo-vm/pull/471, revert it will make everything OK. But it's useful to prevent DDoS.
@devhawk @shargon @roman-khimov

### PR DESCRIPTION
**Pull request by erikzhang**
Close #481

### CODE REVIEW

File: src/Neo.VM/ReferenceCounter.cs [hunk 1]
```diff
@@ -30,9 +33,20 @@ public sealed class ReferenceCounter
         /// </summary>
         public int Count => references_count;
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool NeedTrack(StackItem item)
+        {
+#pragma warning disable CS0162
+            if (TrackAllItems) return true;
```
Comment (shargon) [inline src/Neo.VM/ReferenceCounter.cs:40]: Does we need this private const?

Comment (erikzhang) [reply src/Neo.VM/ReferenceCounter.cs:40]: Keep it for now, the optimizer will take care of it.

Comment (shargon) [reply src/Neo.VM/ReferenceCounter.cs:40]: We can add a `#if DEGUG`

Comment (erikzhang) [reply src/Neo.VM/ReferenceCounter.cs:40]: It's not a debug thing.

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
Test OK from my local to sync mainnet, data comparison is OK. It costs me 6 hours to sync mainnet full data, close to 3.1.0's performance.

**Review by superboyiii — APPROVED** [review]

**Pull request merged**